web_src/js/index.ts

```diff
@@ -9,5 +9,15 @@ import {onDomReady} from './utils/dom.ts';
 import 'htmx.org';
 
 onDomReady(async () => {
-  await import(/* webpackChunkName: "index-domready" */'./index-domready.ts');
+  // when navigate before the import complete, there will be an error from webpack chunk loader:
+  // JavaScript promise rejection: Loading chunk index-domready failed.
+  try {
+    await import(/* webpackChunkName: "index-domready" */'./index-domready.ts');
+  } catch (e) {
+    if (e.name === 'ChunkLoadError') {
+      console.error('Error loading index-domready:', e);
+    } else {
+      throw e;
+    }
+  }
```

Comment:
Side note: So we are waiting for `DOMContentLoaded` until we load this chunk? I think this will cause unnecessary delay and we could start this load earlier, e.g. as soon as this js is parsed.

Reply:
If there are no major benefits to this 2-chunk approach, maybe we should have all main JS in one chunk only.

Reply:
I also think we can merge them into one `index.js` and load it in the `<head>`, total size is about 2.7MB

Reply:
That value seems off. In 1.24 I see much lower sizes:

<img width="902" alt="image" src="https://github.com/user-attachments/assets/22f47294-4fa9-4d48-89c0-ba911b1533e0" />

Reply:
> That value seems off. In 1.24 I see much lower sizes:

If it is, which part makes the new index.js so large? https://github.com/go-gitea/gitea/issues/34960#issuecomment-3038557428

Reply:
Ah yes likely you are seeing dev values. Dev JS is not minified.